connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -880,6 +880,12 @@ ble_error_t Gap::setWhitelist(const whitelist_t &whitelist)
 
 ble_error_t Gap::enablePrivacy(bool enable)
 {
+#if (BLE_GAP_HOST_BASED_PRIVACY == 0)
+    /* we need either privacy on host or controller to enable it */
+    if (enable && !_address_registry.is_controller_privacy_supported()) {
```

Comment:
Why do we check for `enable` if host base privacy is not enabled ?

Reply:
narrowed down the meaning to address resolution and fixed ifdefs accordingly